src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs

```diff
@@ -303,31 +388,48 @@ public FormulaValue GetResult(IRContext irContext)
 
         private class AverageAgg : SumAgg
         {
+            public override FormulaValue NoElementValue(IRContext context)
+            {
+                return CommonErrors.DivByZeroError(context);
+            }
+
             public override FormulaValue GetResult(IRContext irContext)
             {
                 if (_count == 0)
                 {
                     return CommonErrors.DivByZeroError(irContext);
                 }
 
+                if (double.IsInfinity(_accumulator))
+                {
+                    return CommonErrors.OverflowError(irContext);
+                }
+
                 return new NumberValue(irContext, _accumulator / _count);
             }
         }
 
         private static FormulaValue RunAggregator(IAggregator agg, IRContext irContext, FormulaValue[] values)
         {
-            foreach (var value in values)
+            var vals = values.Where(v => v is not BlankValue);
+
+            if (!vals.Any())
+            {
+                return agg.NoElementValue(irContext);
+            }
+
+            foreach (var value in vals)
             {
                 agg.Apply(value);
             }
 
             return agg.GetResult(irContext);
         }
 
-        private static async Task<FormulaValue> RunAggregatorAsync(IAggregator agg, EvalVisitor runner, SymbolContext context, IRContext irContext, FormulaValue[] args)
```

Comment:
Do we need this? 
Or do we get this naturally from just calling getResult() with 0 items?

Reply:
You're right we don't need this.
I've simplified the code here